### ISSUE TITLE
Visualization of enhanced dependencies

### ISSUE THREAD

**Issue by martinpopel** [UniversalDependencies/docs#366]
[Brat](http://universaldependencies.org/visualization.html)+[conllu.js](https://github.com/spyysalo/conllu.js) should provide a nice visualization of CONLL-U v2, including the enhanced dependencies and empty nodes. Brat should do this automatically when using the conllu input format (i.e. without having to enter manually `visual-style` comments). This is related to issues #360 and #363, but @spyysalo suggested it is discussed in a separate issue.

I have two suggestions how Brat could distinguish *basic* and *enhanced* dependencies:
- Enhanced-only deps in red, basic-only deps in blue, basic&enhanced in black. See [an example](http://universaldependencies.org/sandbox.html#pirate-example) (copied in the image below, in case it disappears from the sandbox):
![image](https://cloud.githubusercontent.com/assets/724617/20884765/90cf1838-baed-11e6-8c18-25476b767620.png)

- If the set of enhanced dependencies is not equal to the set of basic dependencies, enhanced dependencies are shown below the sentence.
![image](https://camo.githubusercontent.com/1d070f5b50d1c370b4f755fc9c95468f7bb4c4a5/68747470733a2f2f612e6673646e2e636f6d2f636f6e2f6170702f70726f6a2f74696b7a2d646570656e64656e63792f73637265656e73686f74732f73656d616e746963732e706e672f3138322f313337)

**Comment by spyysalo** [comment]
Dependencies below the sentence would require changing the brat rendering code (not trivial) and would compete with space that's been considered for showing features in one long-standing request.

Automatic coloring would be comparatively easy to do, and I'd be happy to work on this if there's support for the idea. ( @jnivre @dan-zeman , others, what do you think ?)

**Comment by bguil** [comment]
I have some experience with colors (together with dependencies below the sentence).
In LREC 2014, we have proposed an annotation format for French which was called "deep-syntax" (see http://www.lrec-conf.org/proceedings/lrec2014/pdf/494_Paper.pdf).
We also used colors as @martinpopel proposes (with a switch between red and blue): black for common dependencies, red for surface only dependencies, blue for deep only dependencies, moreover blue dependencies are drawn below.
The sequoia French corpus was annotated with this format (see: http://talc2.loria.fr/deep-sequoia/sequoia-7.0/)
I support the use of colors (and of dependencies below the sentence) but I understand that the use of Brat makes the second point difficult.

**Comment by dan-zeman** [comment]
+1 for the automatic coloring using the blue-black-red scheme.

**Comment by spyysalo** [comment]
OK, will do.

**Comment by fginter** [comment]
Thank you for willing to code this in. Just few comments / wishes:
1. Please make it so that these default colors can be overriden using the style
2. Please not red as it evokes "incorrect". I would suggest:
   - Gray for base only (gray evokes - disappearing, less relevant, replaced)
   - Black for base+enhanced
   - Blue for enhanced-only

Thanks

F

**Comment by martinpopel** [comment]
I agree red may not be the best choice as it evokes "incorrect". I've tried gray in #370, but the distinction between black and gray was not clear enough and the goal of #370 was to highlight all the differences between basic and enhanced deps. So, I have chosen **green** for the basic-only deps.
That said, #370 is about the [documentation of enhanced deps](http://universaldependencies.org/u/overview/enhanced-syntax.html); in other places we may not want to highlight the differences and gray may be ok.

Another issue I've encounter is that almost each time there is a preposition or subordinate conjunction or a noun with case feature and deprel nmod, obl, acl and advcl), there will be one basic-only (e.g. "advcl") and one enhanced-only (e.g. "advcl:after") dependency between the same nodes. Thus I suggest that instead of drawing two links, Brat shows just one link and marks the universal part of deprel with black and the added part with blue (e.g. "advcl![<blue>:after</blue>](https://cloud.githubusercontent.com/assets/724617/21057435/ff47ec0a-be39-11e6-8aee-c8293995440d.png)")

**Comment by dan-zeman** [comment]
@fginter : At least for me, red evokes "highlighted", not necessarily "incorrect".

@martinpopel : The two-color label sounds good to me.
